src/app/services/llm/claude.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (C) 2023-2024, Quack AI.
+
+# This program is licensed under the Apache License 2.0.
+# See LICENSE or go to <https://www.apache.org/licenses/LICENSE-2.0> for full license details.
+
+import logging
+from enum import Enum
+from typing import Dict, Generator, List, Union
+
+from anthropic import BaseModel, Client
+
+from .utils import CHAT_PROMPT
+
+logger = logging.getLogger("uvicorn.error")
+
+
+class ClaudeModel(str, Enum):
+    OPUS: str = "claude-3-opus-20240229"
+    SONNET: str = "claude-3-sonnet-20240229"
+    HAIKU: str = "claude-3-haiku-20240307"
+
+
+class ClaudeClient:
+    def __init__(
+        self,
+        api_key: str,
+        model: ClaudeModel,
+        temperature: float = 0.0,
+    ) -> None:
+        self._client = Client(api_key=api_key)
+        self.model = model
+
+        self._validate_model()
+
+        self.temperature = temperature
+        # model_card = BaseModel.retrieve(model)
+        logger.info(
+            f"Using Claude w/ {self.model} (created at "
+            # {datetime.fromtimestamp(model_card.created).isoformat()})",
+        )
+
+    def _validate_model(self) -> None:
```

Comment:
Let's keep it as a static method (with the corresponding arg, like in groq/openai integration)